compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -102,14 +127,13 @@ trait QuotesAndSplices {
         markAsMacro(ctx)
       }
 
-      val (outerQctx, ctx1) = popQuotes()
-
       val internalSplice =
-        outerQctx match
-          case Some(qctxRef) => untpd.Apply(untpd.Apply(untpd.ref(defn.QuotedRuntime_exprNestedSplice.termRef), qctxRef), tree.expr)
-          case _ => untpd.Apply(untpd.ref(defn.QuotedRuntime_exprSplice.termRef), tree.expr)
+        untpd.Apply(untpd.ref(defn.QuotedRuntime_exprSplice.termRef), tree.expr)
```

Comment:
Why is untpd.Apply(... exprSplice ...) still used here as opposed to untpd.Splice?

Reply:
I still use it for the initial typing. For now, I kept it to make sure there are no changes with type inference. This is something I left for later.

Reply:
> This is something I left for later.

OK, in that case maybe a TODO would be appropriate?

Reply:
Added the TODO